### ISSUE TITLE
Docs: Explain the intended usage of respositories and services

### ISSUE THREAD

**Issue by ghferrari** [litestar-org/advanced-alchemy#200]
### Summary

This package looks very useful but it is (as I'm sure you know) missing some essential documentation. In particular, it would be useful to have more documentation about repositories and services and their intended use cases. I've browsed through the `litestar-fullstack` application, which uses both, and it's not especially clear to me what are the different intended use cases of each. Not sure if I'm missing some essential background knowledge here that would make the difference clearer, but if so, it would be nice to have it explained. Thanks for listening!

**Comment by Seemone** [comment]
I agree it's not clear. My take is that repositories handle the DB representation of data on the DB backend, and thus deal with data. Services, in turn, handle what is being done with the data itself, so they are geared towards the controllers

**Comment by ghferrari** [comment]
As a follow up, I've since spent some time with other API frameworks which also distinguish between between repositories and services, so I suspect the authors are to some extent relying on this terminology already being understood.

For those that don't understand it, I'd describe it something like this:

- _Repositories_ are there to provide a suite of relatively low-level database access functions. Essentially they provide methods for pure database queries and updates.
- _Services_ are there to provide higher-level business logic, which at least sometimes might combine multiple repository queries, together with connections to external resources, as well as business logic that isn't captured simply in the database schema.

**Comment by funkindy** [comment]
@ghferrari my understanding is the same. @cofin can you please put a few words, whats your vision on Repository and RepositoryService difference? Do you see any cases where Repository can be used directly?

From my pow they serve more or less the same purposes - CRUD level DB access, but RepositoryService are just little bit more handy.

What is a good way to organize business logic (aka Service) layer with AA abstractions in your opinion?

Thanks!
